Adding providers view using handleItemsList function

heliomar-pena opened the pull request:
## Explanation

- I've added the Providers view. I haven't created other things as: Search bar, Order by filter or Pagination since Daniel Marin haven't confirmed if it is needed, but the code is ready to add it if is needed

- I've added Provider Details view, this include the tabs: history and products that is shown on the [design](https://xd.adobe.com/view/a3457990-d0e6-4f82-876a-f382a7c73582-f616/grid/?hints=off)

- I've changed the layout to do it have the structure as the design (navbar at bottom)

## Screenshots

Providers
![lastscreenshot](https://user-images.githubusercontent.com/66505715/165877309-3e57a813-1d06-4941-ae08-0ebcedbe4916.png)

Provider detail
![lastscreenshot](https://user-images.githubusercontent.com/66505715/165877394-84fdc454-63a8-45ec-9100-89a6770b8100.png)

Provider detail -- products
![lastscreenshot](https://user-images.githubusercontent.com/66505715/165877447-ba3085ee-80fd-45b0-b729-4b016ec1fc6a.png)


## How to test

### How to test providers list view
1. Go to the Preview deploy using this [link](https://deploy-preview-11--did-app.netlify.app/#/providers/)

### How to test Provider detail view
1. Go to the Preview deploy using this [link](https://deploy-preview-11--did-app.netlify.app/#/providers/)
2. Select a provider of the list to go to the Provider Details

### How to test Provider Detail Products view
1. Go to the Preview deploy using this [link](https://deploy-preview-11--did-app.netlify.app/#/providers/)
2. Select a provider of the list to go to the Provider Details
3. Once the provider was charged you could navigate to the Products tab, then await for the products

## Dependencies

- [x] #10

heliomar-pena committed:
Added providers queries

heliomar-pena committed:
Added providers files and path

heliomar-pena committed:
Added providers to navbar

heliomar-pena committed:
Adding provider list logic

heliomar-pena committed:
Added basic styles to Providers list

heliomar-pena committed:
Added state to DOM to show available variables

heliomar-pena committed:
Added input-select component

heliomar-pena committed:
Created list grid component

heliomar-pena committed:
Improving providers view

heliomar-pena committed:
Now is redirecting to providers detail when image is clicked

Israel-Laguan commented:
Hello, just in case I noticed that clicking a provider didn't show the provider info, is that expected?

Israel-Laguan commented on src/actions/ProvidersActions.ts:
```diff
@@ -0,0 +1,45 @@
+          });
+
+          reject(null);
+          return null;
```
Hmm isn't this return unreacheable? Like the scope will finish on the resolve line, please correct me if I am wrong

heliomar-pena replied:
I've thought that too, but without the return the code continue and then I receive the error notification and the success notification

Israel-Laguan replied:
OK, if you tried I am fine, thanks for confirm!

Israel-Laguan commented on src/actions/ProvidersActions.ts:
```diff
@@ -0,0 +1,45 @@
+  offset = 0,
+  query = '',
+}: actionCallbackParamsTypes) => {
+  const name = `%${query}%`;
```
In this case, the name const can be just above variables, as is not used elsewhere

heliomar-pena replied:
Good observation, I'll do it

Israel-Laguan commented on src/layouts/ProviderLayout.vue:
```diff
@@ -0,0 +1,19 @@
+<template>
+  <q-layout view="lHh Lpr lFf">
```
Can I ask why the view has those weird characters ("lHh Lpr lFf")? Does it mean something?

heliomar-pena replied:
`q-layout` component use that to define the layout, that doesn't likes me, I've thought to use another way to define the layout of the APP when I need to add styles to the layout

heliomar-pena replied:
You can use the layout builder https://quasar.dev/layout-builder

Or check this documentation https://quasar.dev/layout/layout

if you want to use "lHh Lpr lFf" and understand what it meant 

![lastscreenshot](https://user-images.githubusercontent.com/66505715/165632230-94d5485d-7a87-4d7a-9002-1f6755531242.png)

Israel-Laguan replied:
Please post in the code some documentation URL.

I agree is weird.

Israel-Laguan commented on src/services/queries/Providers.js:
```diff
@@ -0,0 +1,30 @@
+    $query: String = "%%"
+    $limit: Int = 10
+    $offset: Int = 0
+    $order_by_name: order_by = null
```
maybe make this renaming thing outside the query, please, as the query should be agnostic of such changes (business/code logic)

heliomar-pena replied:
Can you explain a little more please?

Israel-Laguan replied:
```suggestion
    $order_by = null
```

You are renaming here, but I think it make it harder to maintain, you have to remember to change the query as well as the JS code

heliomar-pena replied:
I'll check that possibility, thanks for the suggestions!

heliomar-pena replied:
I will work in that

heliomar-pena replied:
I've fixed it now! 

Now you should pass a object with 

```javascript
{
column: 'name',
content: 'asc'
}
```

for example, then column will be the column that will be used to sort the object on the query. Now we don't need to add a param for each one column, and the frontend can add more columns easily

graphql will receive some as:

```graphql
order_by {
 column: 'asc'
}
```

and will do the search using `column: asc` as order_by

heliomar-pena commented:
> Hello, just in case I noticed that clicking a provider didn't show the provider info, is that expected?

I'll add that functionality soon

heliomar-pena committed:
Moved name const place

heliomar-pena committed:
Exporting ProductsActions from index.js

heliomar-pena committed:
Added Provider Details view

heliomar-pena committed:
Fixing lint warnings

heliomar-pena committed:
fixing lint comments

Israel-Laguan commented on src/pages/Provider/Provider.scss:
```diff
@@ -0,0 +1,97 @@
+  position: relative;
+
+  &__img {
+    max-height: 60vh;
```
Not sure if this works on tablets

heliomar-pena replied:
I've made some changes to do it look more as the design (but I still don't are working in the CSS since I don't know how it should looks in tablet, for example jeje)

![lastscreenshot](https://user-images.githubusercontent.com/66505715/165857403-80c5601d-4a19-4406-a7f9-2598d3fc7059.png)

![lastscreenshot](https://user-images.githubusercontent.com/66505715/165857596-2437eb4c-0395-47a6-865d-081e003dc45f.png)

heliomar-pena committed:
Changed layout

heliomar-pena committed:
Added backBtn

heliomar-pena committed:
Changed order by logic to do more easy to change the column names from the frotend

heliomar-pena committed:
Improved the Typescript logic

heliomar-pena commented:
I've improve a lot the typescript logic and types, I've added types for the database Models (Products, Users, Providers) and I've fixed all lint typescript warnings (warnings about to use "any" as type)

![lastscreenshot](https://user-images.githubusercontent.com/66505715/165991403-38654982-5c4b-491b-b9a4-6c33dd3a64c6.png)
